plugins/widget/plugin.js

```diff
@@ -3256,11 +3485,15 @@
 
 			for ( cl in previousClasses ) {
 				// Avoid removing and adding classes again.
-				if ( !( newClasses && newClasses[ cl ] ) )
+				if ( !( newClasses && newClasses[ cl ] ) ) {
 					this.removeClass( cl );
+					addRemoveClassToStyleDef( this, cl, 1 );
+				}
 			}
-			for ( cl in newClasses )
+			for ( cl in newClasses ) {
 				this.addClass( cl );
+				addRemoveClassToStyleDef( this, cl );
```

Comment:
Shouldn't it be moved into `addClass`?

Reply:
Same as https://github.com/ckeditor/ckeditor-dev/pull/1681#discussion_r183987783